### PR TITLE
Add ability to configure genai to use snapshot instead of thumbnails

### PR DESCRIPTION
**Pull request by hawkeye217**
## Proposed change
Some users have better results with genai reading license plates when higher resolution images are used. This PR adds the ability to generate and regenerate tracked object descriptions via snapshots via a per-camera config option as well as a button in the object details pane in the UI.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

### CODE REVIEW

File: frigate/api/defs/regenerate_query_parameters.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from typing import Optional
+
+from pydantic import BaseModel
+
+
+class RegenerateQueryParameters(BaseModel):
+    source: Optional[str] = "thumbnails"
```
Comment (NickM-27) [inline frigate/api/defs/regenerate_query_parameters.py:7]: this type here should probably use the enum type

Comment (hawkeye217) [reply frigate/api/defs/regenerate_query_parameters.py]: oh for sure

File: frigate/api/defs/regenerate_query_parameters.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from typing import Optional
+
+from pydantic import BaseModel
+
+from frigate.events.types import RegenerateDescriptionEnum
+
+
+class RegenerateQueryParameters(BaseModel):
+    source: Optional[str] = RegenerateDescriptionEnum.thumbnails
```
Comment (NickM-27) [inline frigate/api/defs/regenerate_query_parameters.py:9]:
```suggestion
    source: Optional[RegenerateDescriptionEnum] = RegenerateDescriptionEnum.thumbnails
```

File: frigate/api/defs/regenerate_query_parameters.py (hunk 1 above)
Comment (NickM-27) [reply frigate/api/defs/regenerate_query_parameters.py]: I mean the type as well, so the validation is applied

**Review by NickM-27 — APPROVED** [review]

**Pull request merged**